### DISCUSSION TITLE
Create object from schema with default values

### DISCUSSION THREAD

**Discussion by michalCapo** [ianstormtaylor/superstruct#1253] [Ideas]
This should be as core functionality. Function make should create JS object with default values given by schema.

```
const Some = object({
  age: number(),
  name: string(),
  list: array(string()),
})

const obj = Some.make()

assert(obj.age === Infinity)
assert(obj.name === "")
assert(obj.list.length === 0)

```

**Comment by arturmuller** [comment]
Hi @michalCapo -- thank you for the suggestion!

I think it would be pretty hard to set default values that people would universally agree on. However, you can achieve the equivalent functionality to the above with defaults.

```ts
// Defaults on individual fields
const Some = object({
  age: defaulted(number(), Infinity),
  name: defaulted(string(), ""),
  list: defaulted(array(string()), () => []),
});
Some.create({}) // or create({}, Some)

// Default on the whole object
const Some2 = defaulted(
  object({
    age: number(),
    name: string(),
    list: array(string()),
  }),
  () => ({ age: Infinity, name: "", list: []})
)
Some2.create(undefined) // or create(undefined, Some)
```

We could consider making the the first argument to `Struct#create()` optional, so that in this case, you could just call `Some.create()` without the ugly `undefined` hanging in there.

**Reply by michalCapo** [reply]
Hmm, interesting.

**Reply by yeoffrey** [reply]
I think I agree with @arturmuller here, since the API already supports default arguments this way to me it feels like having default values is a bit redundant.

If you feel this answered your question, can you mark it answered? 😄

**Reply by michalCapo** [reply]
Yes you are right, but we need define defaults for all objects. I was thinking to have general solution to this. But like I sad, it's ok.
